pkg/scheduler/preemption/preemption.go

```diff
@@ -128,40 +128,31 @@ func (p *Preemptor) issuePreemptions(ctx context.Context, targets []*workload.In
 	defer cancel()
 	workqueue.ParallelizeUntil(ctx, parallelPreemptions, len(targets), func(i int) {
 		target := targets[i]
-		patch := workload.BaseSSAWorkload(target.Obj)
-		patch.Status.Conditions = []metav1.Condition{{
-			Type:               kueue.WorkloadAdmitted,
-			Status:             metav1.ConditionFalse,
-			LastTransitionTime: metav1.Now(),
-			Reason:             "Preempted",
-			Message:            "Preempted to accommodate a higher priority Workload",
-		}}
-		err := p.applyPreemption(ctx, patch)
-		if err != nil {
-			errCh.SendErrorWithCancel(err, cancel)
-			return
-		}
-
-		workload.SetEvictedCondition(patch, kueue.WorkloadEvictedByPreemption, "Preempted to accommodate a higher priority Workload")
-		err = workload.ApplyAdmissionStatus(ctx, p.client, patch, false)
-		if err != nil {
-			errCh.SendErrorWithCancel(err, cancel)
-			return
-		}
+		if !apimeta.IsStatusConditionTrue(target.Obj.Status.Conditions, kueue.WorkloadEvicted) {
+			err := p.applyPreemption(ctx, target.Obj)
+			if err != nil {
+				errCh.SendErrorWithCancel(err, cancel)
+				return
+			}
 
-		origin := "ClusterQueue"
-		if cq.Name != target.ClusterQueue {
-			origin = "cohort"
+			origin := "ClusterQueue"
+			if cq.Name != target.ClusterQueue {
+				origin = "cohort"
+			}
+			log.V(3).Info("Preempted", "targetWorkload", klog.KObj(target.Obj))
+			p.recorder.Eventf(target.Obj, corev1.EventTypeNormal, "Preempted", "Preempted by another workload in the %s", origin)
+		} else {
+			log.V(3).Info("Preemption ongoing", "targetWorkload", klog.KObj(target.Obj))
 		}
-		log.V(3).Info("Preempted", "targetWorkload", klog.KObj(target.Obj))
-		p.recorder.Eventf(target.Obj, corev1.EventTypeNormal, "Preempted", "Preempted by another workload in the %s", origin)
 		atomic.AddInt64(&successfullyPreempted, 1)
```

Comment:
It looks a bit weird to me. Does this process count even if the target workload doesn't have `WorkloadEvicted =true` according to this implementation, right?

However, if the workload doesn't have the condition, we don't set the `WorkloadEvictedByPreemption=true` condition to the workload, right?

cc: @alculquicondor

Reply:
Ah, I see. If `!(WorkloadEvicted=true)`, other processes have already preempted that workload.
Thank you!